CMakeLists.txt

```diff
@@ -97,7 +97,9 @@ else()
   phylanx_error("HPX_DIR has not been specified, please set it to help locating HPX")
 endif()
 
-find_package(Eigen3 REQUIRED NO_CMAKE_PACKAGE_REGISTRY)
+find_package(PkgConfig)
+pkg_search_module(Eigen3 REQUIRED eigen3)
```

Comment:
This will not work on platforms not supporting pkgconfig. We could have your code as a fallback, but it will not work as the main detection mechanism.